src/diffusers/models/unet_2d.py

```diff
@@ -144,6 +145,9 @@ def __init__(
         elif time_embedding_type == "positional":
             self.time_proj = Timesteps(block_out_channels[0], flip_sin_to_cos, freq_shift)
             timestep_input_dim = block_out_channels[0]
+        elif time_embedding_type == "learned":
+            self.time_proj = nn.Embedding(num_train_timesteps, block_out_channels[0])
```

Comment:
Should `num_train_timesteps` default to an int value?  What happens if `time_embedding_type` is set to `learned` and no timesteps are provided?

Reply:
I think it's acceptable to assume someone hitting this code path will know to provide the right value, and if not pytorch will throw an error